src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +435,123 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 
+{
+    this.CustomHttpHeaders = new Dictionary <string, string> ();
+
+    if (String.IsNullOrWhiteSpace(stringValue)) {
+        return;
+    }
+
+    if (ValidateParantheses(stringValue) && ValidateFirstLast(stringValue) && ValidateDelimeters(stringValue)) 
+    {
+        string header = stringValue.TrimEnd('}', ' ');
+        string[] elements = header.Split('}');
+        foreach (var element in elements)
+        {
+            string _element = element.TrimStart(',', '{', ' ');
+            string[] array = _element.Split(':');

```

Comment:
the string value can have ":", right? how about it?

Reply:
The input string has key value pairs e.g
"{Accept:application/json},{Authorization: Bearer xxxyssh-token}";